### PR TITLE
Fix version comparison for Python 3.10 and 4

### PR DESCRIPTION
**Pull request by hugovk**
### Fix for Python 3.10: use `sys.version_info` instead of comparing `sys.version` to string

Don't assume the minor version is a single character: `'3.10' >= '3.4'` is `False`!

---

### Fix for Python 4: replace unsafe `six.PY3` with `six.PY2`

There's some code which uses `six.PY3`:

```python
if six.PY3:
    print("Python 3+ code")
else:
    print "Python 2 code" 
```

Where:

```python
PY3 = sys.version_info[0] == 3
```

When run on Python 4, this will run the Python 2 code!

Instead, use `six.PY2`.

---

Found using https://github.com/asottile/flake8-2020.

### CODE REVIEW

**Review by mdanatg — APPROVED** [review]

File: tensorflow/tools/docs/generate_lib.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def write_docs(output_dir,
```
Comment (mdanatg) [inline tensorflow/tools/docs/generate_lib.py]: Sorry for the delay. It looks like the docs tool is being moved to a separate package, and the owners prefer not to make changes until that's complete. Could you remove this file only?

**Review by mdanatg — CHANGES_REQUESTED** [review]

File: tensorflow/tools/docs/generate_lib.py (hunk 1 above)
Comment (hugovk) [reply tensorflow/tools/docs/generate_lib.py]: Removed!

Comment (mdanatg) [reply tensorflow/tools/docs/generate_lib.py]: Thanks! It still shows in the PR, as deleted. We need to exclude any changes to it from the PR.

Comment (hugovk) [reply tensorflow/tools/docs/generate_lib.py]: How do you mean, exactly? Squash the two commits to one?

Comment (mdanatg) [reply tensorflow/tools/docs/generate_lib.py]: I believe `git reset` should work. `git revert` + squash might work too.

Comment (hugovk) [reply tensorflow/tools/docs/generate_lib.py]: How does it look now?

Comment (mdanatg) [reply tensorflow/tools/docs/generate_lib.py]: Still shows deleted, but maybe the UI is confusing me. Let's try to pull it and see if the tests are happy.

**Review by mdanatg — DISMISSED** [review]

File: tensorflow/tools/docs/generate_lib.py (hunk 1 above)
Comment (hugovk) [reply tensorflow/tools/docs/generate_lib.py]: I'm a bit confused, when you said to remove `tensorflow/tools/docs/generate_lib.py`, I understood you meant to delete it.

Did you instead mean to not make any changes to that file in this PR?

Thanks!

Comment (mdanatg) [reply tensorflow/tools/docs/generate_lib.py]: Yes, sorry for the confusion. I guess I should have said to revert it.

Comment (hugovk) [reply tensorflow/tools/docs/generate_lib.py]: No problem at all! How's it look now?

Comment (mdanatg) [reply tensorflow/tools/docs/generate_lib.py]: Looks good now, thank you!

**Review by mdanatg — APPROVED** [review]

**Pull request merged**